machine-learning/app/main.py

```diff
@@ -103,7 +111,9 @@ async def predict(
     except orjson.JSONDecodeError:
         raise HTTPException(400, f"Invalid options JSON: {options}")
 
-    model = await load(await model_cache.get(model_name, model_type, **kwargs))
+    key = f"{model_name}{model_type.value}{kwargs.get('mode', '')}"
+    ttl = settings.model_ttl if key in model_cache.cache._handlers else None
```

Comment:
This gets handled in the model cache

Reply:
Right, I was worried about it resetting the ttl but `load` only happens for models which aren't loaded, so not the preloaded ones, thanks